### ISSUE TITLE
Intercept function calls from the global namespace

### ISSUE THREAD

**Issue by lisachenko** [goaop/framework#53]
According to the official documentation http://www.php.net/manual/en/language.namespaces.rules.php there is a name resolution rule:

> Inside namespace (say A\B), calls to unqualified functions are resolved at run-time. Here is how a call to function foo() is resolved:
> 
>   It looks for a function from the current namespace: A\B\foo().
>   It tries to find and call the global function foo().

This rule can be used by Go! library to dynamically intercept global function calls in the namespaces. For example, if we want to intercept a function `file_get_contents()` in the application, we can create this function in the current namespace, so it will be used instead of global function.

**Comment by lisachenko** [comment]
Example of mocking `usleep()` function:

``` php
use Go\Aop\Framework\FunctionAroundInterceptor;
use Go\Aop\Framework\ReflectionFunctionInvocation;
use Go\Aop\Intercept\FunctionInvocation;


function usleep($microseconds)
{
    $interceptor = new FunctionAroundInterceptor(function (FunctionInvocation $invocation) {
        echo 'Hello!';
        return $invocation->proceed();
    });
    $joinpoint = new ReflectionFunctionInvocation('usleep', array($interceptor));
    return $joinpoint(array($microseconds));
}
```

**Comment by lisachenko** [comment]
Merged in 28e5e81faf0d60dee670504fdddd8545c4824162
